From 0bb4b8b57fb98b40d216351ffe26f2d8fd699642 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Albert=20Bab=C3=AD?= <albert.babi@savoirfairelinux.com>
Date: Fri, 8 Oct 2021 09:59:17 +0200
Subject: [PATCH] disable libx11 dependency

This dependency is irrelevant on i.MX6 platforms because we don't use X11.
Moreover compiling with libX11 requires to enable the disto feature X11
which will change the compilation of Qt5 and disable the EGLFS_VIV
plugin.

---
 CMakeLists.txt    |  6 +-----
 src/avadapter.cpp | 21 ---------------------
 2 files changed, 1 insertion(+), 26 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7f22cf5..3a97ca3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -228,11 +228,9 @@ if(MSVC)
                         ${QRENCODE})
 else()
     list(APPEND COMMON_SOURCES
-                ${SRC_DIR}/xrectsel.c
                 ${SRC_DIR}/dbuserrorhandler.cpp)
     list(APPEND COMMON_HEADERS
-                ${SRC_DIR}/dbuserrorhandler.h
-                ${SRC_DIR}/xrectsel.h)
+                ${SRC_DIR}/dbuserrorhandler.h)
     list(APPEND QML_LIBS Qt5::DBus)
     list(APPEND QML_LIBS_LIST DBus)
 
@@ -307,7 +305,6 @@ else()
 
     find_library(ringclient ringclient ${LRCLIBDIR} NO_DEFAULT_PATH)
     find_library(qrencode qrencode)
-    find_library(X11 X11)
 endif()
 
 # Qt find package
@@ -423,7 +420,6 @@ else()
                           ${QML_LIBS}
                           ${LRC_LIB_NAME}
                           ${qrencode}
-                          ${X11}
                           ${LIBNM_LIBRARIES}
                           ${LIBNOTIFY_LIBRARIES}
                           ${LIBGDKPIXBUF_LIBRARIES}
diff --git a/src/avadapter.cpp b/src/avadapter.cpp
index ef3f8b0..d580add 100644
--- a/src/avadapter.cpp
+++ b/src/avadapter.cpp
@@ -24,10 +24,6 @@
 #include "api/newcodecmodel.h"
 #include "api/newdevicemodel.h"
 
-#ifdef Q_OS_LINUX
-#include "xrectsel.h"
-#endif
-
 #include <QtConcurrent/QtConcurrent>
 #include <QApplication>
 #include <QPainter>
@@ -175,29 +171,12 @@ AvAdapter::shareFile(const QString& filePath)
 void
 AvAdapter::shareScreenArea(unsigned x, unsigned y, unsigned width, unsigned height)
 {
-#ifdef Q_OS_LINUX
-    // xrectsel will freeze all displays too fast so that the call
-    // context menu will not be closed even closed signal is emitted
-    // use timer to wait until popup is closed
-    QTimer::singleShot(100, [=]() mutable {
-        x = y = width = height = 0;
-        xrectsel(&x, &y, &width, &height);
-
-        lrcInstance_->avModel().setDisplay(getScreenNumber(),
-                                           x,
-                                           y,
-                                           width < 128 ? 128 : width,
-                                           height < 128 ? 128 : height,
-                                           lrcInstance_->getCurrentCallId());
-    });
-#else
     lrcInstance_->avModel().setDisplay(getScreenNumber(),
                                        x,
                                        y,
                                        width < 128 ? 128 : width,
                                        height < 128 ? 128 : height,
                                        lrcInstance_->getCurrentCallId());
-#endif
 }
 
 void
-- 
2.25.1

